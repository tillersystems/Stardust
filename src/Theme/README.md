# Theme

### Usage

```jsx
import { Theme } from '@tillersystems/stardust';
```

<!-- STORY -->
