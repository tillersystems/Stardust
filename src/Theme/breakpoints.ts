/**
 * Defines application breakpoints.
 */

const xs = 0;
const sm = 576;
const md = 768;
const lg = 992;
const xl = 1200;

export const breakpoints = {
  xs,
  sm,
  md,
  lg,
  xl,
};
