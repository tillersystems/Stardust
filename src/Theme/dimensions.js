/**
 * Defines the theming dimensions.
 */

const radius = 0.4;
const tiny = 0.3;
const small = 0.7;
const medium = 1.5;
const big = 3.1;
const huge = 4.5;

export const dimensions = {
  radius: `${radius}rem`,

  tiny: `${tiny}rem`,
  small: `${small}rem`,
  medium: `${medium}rem`,
  big: `${big}rem`,
  huge: `${huge}rem`,

  radiusInt: radius,

  tinyInt: tiny,
  smallInt: small,
  mediumInt: medium,
  bigInt: big,
  hugeInt: huge,
};
