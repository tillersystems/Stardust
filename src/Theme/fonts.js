/**
 * Defines the theming of fonts.
 */

export const fonts = {
  size: {
    // -start- Old font size
    tiny: '0.3rem',
    small: '0.7rem',
    // medium: '1.2rem',
    // big: '3.2rem',
    huge: '6.4rem',
    // -end- Old font size

    // -start- New font size
    default: '1.2rem',
    medium: '1.4rem',
    big: '1.6rem',
    h6: '1.8rem',
    h5: '2.0rem',
    h4: '2.2rem',
    h3: '2.6rem',
    h2: '3rem',
    h1: '3.6rem',
    // -end- New font size

    // -start- Old font int
    tinyInt: 0.3,
    smallInt: 0.7,
    // mediumInt: 1.2,
    // bigInt: 3.2,
    hugeInt: 6.4,
    // -end- Old font int

    // -start- Old font int
    defaultInt: 1.2,
    mediumInt: 1.4,
    bigInt: 1.6,
    h6Int: 1.8,
    h5Int: 2.0,
    h4Int: 2.2,
    h3Int: 2.6,
    h2Int: 3,
    h1Int: 3.6,
    // -end- Old font int
  },

  spacing: {
    small: '0.5px',
    medium: '1px',
    big: '1.5px',
  },

  weight: {
    thin: 200,
    normal: 400,
    thick: 600,
  },

  family: '"Open Sans", Helvetica, Arial, sans-serif',
};
