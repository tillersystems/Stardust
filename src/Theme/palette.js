/**
 * Defines the theming colors.
 */

import { lighten, darken } from 'polished';

const dark = color => darken(0.03, color);

const light = color => lighten(0.1, color);

// -start- Old colours palette
const white = 'hsl(0, 100%, 100%)';
const gray = 'hsl(218, 18%, 91%)';
const black = 'hsl(0, 0%, 0%)';
const blackLight = 'hsl(213, 17%, 20%)';

// const red = 'hsl(6, 79%, 65%)';
// const green = 'hsl(89, 53%, 52%)';
// const blue = 'hsl(200, 74%, 46%)';
const orange = 'hsl(34, 91%, 65%)';
const anthracite = 'hsl(213, 17%, 20%)';
const clay = 'hsl(215, 17%, 26%)';
const sand = 'hsl(214, 8%, 84%)';
const marble = 'hsl(206, 23%, 69%)';
const grey = 'hsl(206, 16%, 60%)';
const bodyBackground = 'hsl(218, 29%, 95%)';
const emerald = 'hsl(145, 63%, 49%)';
const river = 'hsl(204, 70%, 53%)';
const wisteria = 'hsl(282, 44%, 47%)';
const carrot = 'hsl(28, 80%, 52%)';
const alizarin = 'hsl(6, 78%, 57%)';
const turquoise = 'hsl(168, 76%, 42%)';
const greygoose = 'hsl(216, 29%, 97%)';
const fadeBlue = 'hsl(210, 50%, 98%)';

const forePrimary = white;
// const backPrimary = blue;

const foreSecondary = grey;
const backSecondary = white;

const foreSuccess = white;
// const backSuccess = green;

const foreFailure = white;
// const backFailure = red;

const googleBrandRed = '#ea4335';
// -end- Old colours palette

// -start- New colours palette
const whiteOpacity = opacity => `hsla(0, 100%, 100%, ${opacity})`;

const red = 'hsl(6, 79%, 65%)';
const red1 = 'hsl(6, 60%, 58%)';
const red2 = 'hsl(6, 52%, 55%)';
const green = 'hsl(89, 53%, 52%)';
const green1 = 'hsl(89, 49%, 45%)';
const green2 = 'hsl(89, 49%, 42%)';
const blue = 'hsl(200, 74%, 46%)';
const blue1 = 'hsl(200, 74%, 41%)';
const blue2 = 'hsl(200, 74%, 38%)';
const darkBlue = 'hsl(213, 17%, 20%)';
const yellow = 'hsl(9, 91%, 65%)';

const mysticGrey = 'hsl(216,45%,98%)';
const lightGrey = 'hsl(207,22%,90%)';
const mediumGrey = 'hsl(206,23%,69%)';
const darkGrey = 'hsl(206,16%,60%)';
const spaceGrey = 'hsl(207,13%,45%)';

const backPrimary = blue;
const backFailure = red;
const backSuccess = green;
// -end- New colours palette

export const palette = {
  // -start- Old colours palette
  white,
  black,
  blackLight,

  googleBrandRed,

  gray,
  greygoose,
  darkGray: dark(gray),
  lightGray: light(gray),

  red,
  darkRed: dark(red),
  lightRed: light(red),

  green,
  darkGreen: dark(green),
  lightGreen: light(green),

  blue,
  // darkBlue: dark(blue),
  lightBlue: light(blue),

  orange,

  anthracite,
  clay,
  sand,
  marble,
  emerald,
  river,
  wisteria,
  carrot,
  alizarin,
  turquoise,
  grey,
  bodyBackground,
  fadeBlue,

  primary: {
    fore: forePrimary,
    darkFore: dark(forePrimary),
    lightFore: light(forePrimary),

    back: backPrimary,
    darkBack: dark(backPrimary),
    lightBack: light(backPrimary),

    default: blue,
    dark: blue1,
    darker: blue2,
  },

  secondary: {
    fore: foreSecondary,
    darkFore: dark(foreSecondary),
    lightFore: light(foreSecondary),

    back: backSecondary,
    darkBack: dark(backSecondary),
    lightBack: light(backSecondary),
  },

  success: {
    fore: foreSuccess,
    darkFore: dark(foreSuccess),
    lightFore: light(foreSuccess),

    back: backSuccess,
    darkBack: dark(backSuccess),
    lightBack: light(backSuccess),

    default: green,
    dark: green1,
    darker: green2,
  },

  failure: {
    fore: foreFailure,
    darkFore: dark(foreFailure),
    lightFore: light(foreFailure),

    back: backFailure,
    darkBack: dark(backFailure),
    lightBack: light(backFailure),

    default: red,
    dark: red1,
    darker: red2,
  },
  // -end- Old colours palette

  // -start- New colours palette
  whiteOpacity,
  // red,
  // green,
  // blue,
  darkBlue,
  yellow,

  mysticGrey,
  lightGrey,
  mediumGrey,
  darkGrey,
  spaceGrey,
  // -end- New colours palette
};
