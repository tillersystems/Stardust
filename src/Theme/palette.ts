/**
 * Defines the theming colors.
 */
import { whiteOpacity, setColorDark } from '../helpers/colors';
// Black and white colors
const white = 'hsl(0,100%,100%)';
const black = 'hsl(0,0%,0%)';

// Avatar colors
const emerald = 'hsl(145,63%,49%)';
const river = 'hsl(204,70%,53%)';
const wisteria = 'hsl(282,44%,47%)';
const carrot = 'hsl(28,80%,52%)';
const alizarin = 'hsl(6,78%,57%)';
const turquoise = 'hsl(168,76%,42%)';

// Google color
const googleBrandRed = 'hsl(5,81%,56%)';

// Shades of grey
const mysticGrey = 'hsl(218, 29%, 95%)';
const lightGrey = 'hsl(207,22%,90%)';
const mediumGrey = 'hsl(206,23%,69%)';
const darkGrey = 'hsl(206,16%,60%)';
const spaceGrey = 'hsl(207,13%,45%)';
const paleGrey = 'hsl(240,33%,99%)';
const veryLightBlue = 'hsl(220,27%,94%)';
//Temporary name
const veryLightGrey = 'hsl(214,64%,98%)';

// Primary colors
const blue = 'hsl(200,74%,46%)';
const green = 'hsl(89,53%,52%)';
const red = 'hsl(6,79%,65%)';
const yellow = 'hsl(34,91%,65%)';
const bodyBackground = paleGrey;
const darkBlue = 'hsl(213,17%,20%)';

export const palette = {
  // Black and white colors
  white,
  whiteOpacity,
  black,

  // Avatar colors
  avatar: {
    emerald,
    river,
    wisteria,
    carrot,
    alizarin,
    turquoise,
  },

  // Google color
  googleBrandRed,

  // Primary colors
  bodyBackground,
  primary: {
    default: blue,
    dark: setColorDark(blue, 0.1),
    darker: setColorDark(blue, 0.15),
  },
  secondary: {
    default: white,
  },
  success: {
    default: green,
    dark: setColorDark(green, 0.1),
    darker: setColorDark(green, 0.15),
  },
  failure: {
    default: red,
    dark: setColorDark(red, 0.1),
    darker: setColorDark(red, 0.15),
  },
  warning: {
    default: yellow,
    dark: setColorDark(yellow, 0.1),
    darker: setColorDark(yellow, 0.15),
  },
  darkBlue,
  veryLightBlue,

  // Shades of grey
  mysticGrey,
  lightGrey,
  mediumGrey,
  darkGrey,
  spaceGrey,
  paleGrey,
  veryLightGrey,
};
