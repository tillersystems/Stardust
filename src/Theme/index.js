/**
 * Theme exported to the provider
 * This is globally available in styled-components when interpolating a function like so:
 * ${(props) => props.theme}
 * Or using import { withTheme } from 'styled-components';
 */

import { breakpoints } from './breakpoints';
import { palette } from './palette';
import { dimensions } from './dimensions';
import { fonts } from './fonts';

const Theme = {
  breakpoints,
  palette,
  dimensions,
  fonts,
};

export default Theme;
