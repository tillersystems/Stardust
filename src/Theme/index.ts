import { DefaultTheme } from 'styled-components';

/**
 * Theme exported to the provider
 * This is globally available in styled-components when interpolating a function like so:
 * ${(props) => props.theme}
 * Or using import { withTheme } from 'styled-components';
 */

import { breakpoints } from './breakpoints';
import { palette } from './palette';
import { dimensions } from './dimensions';
import { fonts } from './fonts';

const Theme: DefaultTheme = {
  breakpoints,
  palette,
  dimensions,
  fonts,
};

export default Theme;
