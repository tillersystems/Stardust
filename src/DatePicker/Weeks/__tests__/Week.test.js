import React from 'react';
import Mockdate from 'mockdate';
import { DateTime, Settings } from 'luxon';
import { fireEvent } from '@testing-library/react';
import { transparentize } from 'polished';

import Week from '..';
import Theme from '../../../Theme';

Settings.defaultZoneName = 'utc';

describe('<Weeks />', () => {
  const dateValue = DateTime.fromObject({
    year: 2018,
    month: 7,
    day: 15,
    hour: 0,
    minute: 0,
    second: 0,
  });

  beforeEach(() => {
    // Mock date https://www.npmjs.com/package/mockdate
    // Mockdate.set(date, [timezoneOffset]);
    Mockdate.set(dateValue, 120);
  });

  afterEach(() => {
    // Reset mockdate
    Mockdate.reset();
  });

  test('should render without a problem', () => {
    const { container } = render(<Week currentDate={dateValue} selected={dateValue} />);

    expect(container.firstChild).toMatchSnapshot();
  });

  test('should select a range date', () => {
    const startDate = DateTime.fromObject({
      year: 2018,
      month: 7,
      day: 16,
      hour: 0,
      minute: 0,
      second: 0,
    });
    const endDate = DateTime.fromObject({
      year: 2018,
      month: 7,
      day: 18,
      hour: 0,
      minute: 0,
      second: 0,
    });
    const { getByText } = render(
      <Week currentDate={dateValue} selected={dateValue} startDate={startDate} endDate={endDate} />,
    );

    const startDayNode = getByText('16');
    const endDayNode = getByText('18');

    expect(startDayNode).toHaveStyleRule('color', Theme.palette.white);
    expect(startDayNode).toHaveStyleRule('background', Theme.palette.primary.default);
    expect(startDayNode).not.toHaveStyleRule(
      'background',
      transparentize(0.86, Theme.palette.primary.default),
    );
    expect(startDayNode).not.toHaveStyleRule('background', {
      modifier: ':hover',
    });

    expect(endDayNode).toHaveStyleRule('color', Theme.palette.white);
    expect(endDayNode).toHaveStyleRule('background', Theme.palette.primary.default);
    expect(endDayNode).not.toHaveStyleRule(
      'background',
      transparentize(0.86, Theme.palette.primary.default),
    );
    expect(endDayNode).not.toHaveStyleRule('background', {
      modifier: ':hover',
    });
  });

  test('should handle onDateClick', () => {
    const dateSelected = DateTime.fromObject({
      year: 2018,
      month: 7,
      day: 20,
      hour: 0,
      minute: 0,
      second: 0,
    });
    const spy = jest.fn();
    const { getByText } = render(
      <Week currentDate={dateValue} selected={dateValue} onDateClick={spy} />,
    );

    const dayNode = getByText('20');

    // Click on a day
    fireEvent.click(dayNode);

    expect(spy).toHaveBeenCalledTimes(1);
    expect(spy).toHaveBeenCalledWith(dateSelected);
  });

  test('should not handle onDateClick if the selected day is before min date', () => {
    const spy = jest.fn();
    const { getByText } = render(
      <Week
        currentDate={dateValue}
        selected={dateValue}
        onDateClick={spy}
        selectedDate={dateValue}
        minDate={dateValue}
      />,
    );

    const dayNode = getByText('10');

    // Click on a day
    fireEvent.click(dayNode);

    expect(spy).toHaveBeenCalledTimes(0);
  });

  test('should not handle onDateClick if the selected day is after max date', () => {
    const spy = jest.fn();
    const { getByText } = render(
      <Week currentDate={dateValue} selected={dateValue} onDateClick={spy} maxDate={dateValue} />,
    );

    const dayNode = getByText('20');

    // Click on a day
    fireEvent.click(dayNode);

    expect(spy).toHaveBeenCalledTimes(0);
  });
});
