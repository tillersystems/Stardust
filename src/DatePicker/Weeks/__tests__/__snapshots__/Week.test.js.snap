// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Weeks /> should render without a problem 1`] = `
.c1 {
  margin-top: 0.8rem;
  font-size: 1.4rem;
  color: hsl(213,17%,20%);
  cursor: pointer;
}

.c1:not(:nth-child(7n)) {
  padding-right: 0.8rem;
}

.c4 {
  margin-top: 0.8rem;
  font-size: 1.4rem;
  color: hsl(213,17%,20%);
  border-top-left-radius: 2.8rem;
  border-bottom-left-radius: 2.8rem;
  border-top-right-radius: 2.8rem;
  border-bottom-right-radius: 2.8rem;
  cursor: pointer;
}

.c4:not(:nth-child(7n)) {
  padding-right: 0.8rem;
}

.c4:not(:nth-child(7n)) {
  padding-right: 0;
  margin-right: 0.8rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-basis: 2.8rem;
  -ms-flex-preferred-size: 2.8rem;
  flex-basis: 2.8rem;
  width: 2.8rem;
  height: 2.8rem;
  max-width: 2.8rem;
  border-radius: 2.8rem;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c2:hover {
  background: rgba(30,146,204,0.14);
}

.c2,
.c2:hover {
  background: transparent;
  color: hsl(206,23%,69%);
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-basis: 2.8rem;
  -ms-flex-preferred-size: 2.8rem;
  flex-basis: 2.8rem;
  width: 2.8rem;
  height: 2.8rem;
  max-width: 2.8rem;
  border-radius: 2.8rem;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c3:hover {
  background: rgba(30,146,204,0.14);
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-basis: 2.8rem;
  -ms-flex-preferred-size: 2.8rem;
  flex-basis: 2.8rem;
  width: 2.8rem;
  height: 2.8rem;
  max-width: 2.8rem;
  border-radius: 2.8rem;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c5:hover {
  background: rgba(30,146,204,0.14);
}

.c5,
.c5:hover {
  color: hsl(0,100%,100%);
  background: hsl(200,74%,46%);
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      25
    </div>
  </div>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      26
    </div>
  </div>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      27
    </div>
  </div>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      28
    </div>
  </div>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      29
    </div>
  </div>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      30
    </div>
  </div>
  <div
    class="c1"
  >
    <div
      class="c3"
    >
      1
    </div>
  </div>
  <div
    class="c1"
  >
    <div
      class="c3"
    >
      2
    </div>
  </div>
  <div
    class="c1"
  >
    <div
      class="c3"
    >
      3
    </div>
  </div>
  <div
    class="c1"
  >
    <div
      class="c3"
    >
      4
    </div>
  </div>
  <div
    class="c1"
  >
    <div
      class="c3"
    >
      5
    </div>
  </div>
  <div
    class="c1"
  >
    <div
      class="c3"
    >
      6
    </div>
  </div>
  <div
    class="c1"
  >
    <div
      class="c3"
    >
      7
    </div>
  </div>
  <div
    class="c1"
  >
    <div
      class="c3"
    >
      8
    </div>
  </div>
  <div
    class="c1"
  >
    <div
      class="c3"
    >
      9
    </div>
  </div>
  <div
    class="c1"
  >
    <div
      class="c3"
    >
      10
    </div>
  </div>
  <div
    class="c1"
  >
    <div
      class="c3"
    >
      11
    </div>
  </div>
  <div
    class="c1"
  >
    <div
      class="c3"
    >
      12
    </div>
  </div>
  <div
    class="c1"
  >
    <div
      class="c3"
    >
      13
    </div>
  </div>
  <div
    class="c1"
  >
    <div
      class="c3"
    >
      14
    </div>
  </div>
  <div
    class="c4"
  >
    <div
      class="c5"
    >
      15
    </div>
  </div>
  <div
    class="c1"
  >
    <div
      class="c3"
    >
      16
    </div>
  </div>
  <div
    class="c1"
  >
    <div
      class="c3"
    >
      17
    </div>
  </div>
  <div
    class="c1"
  >
    <div
      class="c3"
    >
      18
    </div>
  </div>
  <div
    class="c1"
  >
    <div
      class="c3"
    >
      19
    </div>
  </div>
  <div
    class="c1"
  >
    <div
      class="c3"
    >
      20
    </div>
  </div>
  <div
    class="c1"
  >
    <div
      class="c3"
    >
      21
    </div>
  </div>
  <div
    class="c1"
  >
    <div
      class="c3"
    >
      22
    </div>
  </div>
  <div
    class="c1"
  >
    <div
      class="c3"
    >
      23
    </div>
  </div>
  <div
    class="c1"
  >
    <div
      class="c3"
    >
      24
    </div>
  </div>
  <div
    class="c1"
  >
    <div
      class="c3"
    >
      25
    </div>
  </div>
  <div
    class="c1"
  >
    <div
      class="c3"
    >
      26
    </div>
  </div>
  <div
    class="c1"
  >
    <div
      class="c3"
    >
      27
    </div>
  </div>
  <div
    class="c1"
  >
    <div
      class="c3"
    >
      28
    </div>
  </div>
  <div
    class="c1"
  >
    <div
      class="c3"
    >
      29
    </div>
  </div>
  <div
    class="c1"
  >
    <div
      class="c3"
    >
      30
    </div>
  </div>
  <div
    class="c1"
  >
    <div
      class="c3"
    >
      31
    </div>
  </div>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      1
    </div>
  </div>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      2
    </div>
  </div>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      3
    </div>
  </div>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      4
    </div>
  </div>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      5
    </div>
  </div>
</div>
`;
